Lib/Protocols/IdFTP.pas

```diff
@@ -2771,29 +2771,30 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-  if FClientInfo.ClientName <> '' then begin
-    if IsExtSupported('CSID') then begin {do not localize}
-
-      SendCmd('CSID ' + FClientInfo.GetCSIDOutput);  {do not localize}
+  if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := FClientInfo.GetCSIDOutput;
+      if LClnt = '' then begin
+        LClnt := 'Name='+gsIdProductName + '; Version=' + gsIdVersion;

```

Comment:
The more I think about it, use of `gsIdProductName` and `gsIdVersion` should probably be moved inside of `GetCSIDOutput()` as the default return value when `FClientName` is empty.